climada/engine/impact.py

```diff
@@ -48,6 +47,410 @@
+
+        Returns
+        -------
+        np.array
+            The cover per exposure point
+
+        """
+        return self.exposures.gdf['cover']
+
+    def impact(self, save_mat=True):
+        """Compute the impact of a hazard on exposures.
+
+        Parameters
+        ----------
+        save_mat : bool
+            if true, save the total impact matrix (events x exposures)
+
+        Examples
+        --------
+            >>> haz = Hazard.from_mat(HAZ_DEMO_MAT)  # Set hazard
+            >>> impfset = ImpactFuncSet.from_excel(ENT_TEMPLATE_XLS) # Set impact functions
+            >>> exp = Exposures(pd.read_excel(ENT_TEMPLATE_XLS)) # Set exposures
+            >>> impcalc = ImpactCal(exp, impfset, haz)
+            >>> imp = impcalc.insured_impact()
+            >>> imp.aai_agg
+
+        Note
+        ----
+        Deductible and/or cover values in the exposures are ignored.
+        """
+        impf_col = self.exposures.get_impf_column(self.hazard.haz_type)
+        exp_gdf = self.minimal_exp_gdf(impf_col)
+        LOGGER.info('Calculating impact for %s assets (>0) and %s events.',
+                    self.n_events, self.n_events)
+        imp_mat_gen = self.imp_mat_gen(exp_gdf, impf_col)
+        if save_mat:
+            self.imp_mat = self.stitch_impact_matrix(imp_mat_gen)
+            at_event, eai_exp, aai_agg = self.risk_metrics()
+        else:
+            at_event, eai_exp, aai_agg = self.stitch_risk_metrics(imp_mat_gen)
+        return Impact.from_eih(
+            self.exposures, self.impfset, self.hazard,
+            at_event, eai_exp, aai_agg, self.imp_mat
+            )
+
+    def insured_impact(self, save_mat=False):
+        """Compute the impact of a hazard on exposures with a deductible and/or
+        cover.
+
+        For each exposure point, the impact per event is obtained by
+        substracting the deductible (and is maximally equal to the cover).
+
+        Parameters
+        ----------
+        save_mat : bool
+            if true, save the total impact matrix (events x exposures)
+
+        Examples
+        --------
+            >>> haz = Hazard.from_mat(HAZ_DEMO_MAT)  # Set hazard
+            >>> impfset = ImpactFuncSet.from_excel(ENT_TEMPLATE_XLS) # Set impact functions
+            >>> exp = Exposures(pd.read_excel(ENT_TEMPLATE_XLS)) # Set exposures
+            >>> impcalc = ImpactCal(exp, impfset, haz)
+            >>> imp = impcalc.insured_impact()
+            >>> imp.aai_agg
+
+        See also
+        --------
+        apply_deductible_to_mat:
+            apply deductible to impact matrix
+        apply_cover_to_mat:
+            apply cover to impact matrix
+        """
+        impf_col = self.exposures.get_impf_column(self.hazard.haz_type)
+        exp_gdf = self.minimal_exp_gdf(impf_col)
+        LOGGER.info('Calculating impact for %s assets (>0) and %s events.',
+                    exp_gdf.size, self.hazard.size)
+
+        if self.imp_mat.size == 0:
```

Comment:
See my other comment
```suggestion
        if self.imp_mat is None:
```